### ISSUE TITLE
natsSubscription_Destroy submits UNSUB even after DELSUB

### ISSUE THREAD

**Issue by Kazmirchuk** [nats-io/nats.c#558]
Hello,
I'm working on a C++ wrapper over cnats, and trying to implement async request/reply (simple "old style", for now)
Using NATS CLI as a responder.
I run `nats-server -DV` and `nats reply --count=1 foo bla` and then my program that in essence does this:
```
natsInbox_Create(&inbox);
natsConnection_SubscribeTimeout(&subscription, ... &asyncRequestCallback...;
natsSubscription_AutoUnsubscribe(subscription, 1));
natsConnection_PublishRequest(...);

in asyncRequestCallback:
natsMsg_Destroy ...
natsSubscription_Destroy ...
```
And these are the logs from nats-server (with PING/PONG chatter removed):
```
cid:4 - Client connection created
cid:4 - <<- [CONNECT {"verbose":false,"pedantic":false,"tls_required":false,"name":"NATS CLI Version 0.0.33","lang":"go","version":"1.16.0","protocol":1,"echo":true,"headers":true,"no_responders":true}]
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - <<- [SUB foo NATS-RPLY-22 1]
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - <<- [UNSUB 1 1]
cid:5 - Client connection created
cid:5 - <<- [CONNECT {"verbose":false,"pedantic":false,"tls_required":false,"name":"","lang":"C","version":"3.1.1","protocol":1,"echo":true,"headers":true,"no_responders":false}]
cid:5 - "v3.1.1:C" - <<- [SUB _INBOX.0MNLNJTAGRZNY6RK4P22WK  1]
cid:5 - "v3.1.1:C" - <<- [UNSUB 1 1]
cid:5 - "v3.1.1:C" - <<- [PUB foo _INBOX.0MNLNJTAGRZNY6RK4P22WK 3]
cid:5 - "v3.1.1:C" - <<- MSG_PAYLOAD: ["bar"]
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - Auto-unsubscribe limit of 1 reached for sid '1'
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - ->> [MSG foo 1 _INBOX.0MNLNJTAGRZNY6RK4P22WK 3]
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - <-> [DELSUB 1]
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - <<- [PUB _INBOX.0MNLNJTAGRZNY6RK4P22WK 3]
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - <<- MSG_PAYLOAD: ["bla"]
cid:5 - "v3.1.1:C" - Auto-unsubscribe limit of 1 reached for sid '1'
cid:5 - "v3.1.1:C" - ->> [MSG _INBOX.0MNLNJTAGRZNY6RK4P22WK 1 3]
cid:5 - "v3.1.1:C" - <-> [DELSUB 1]
cid:5 - "v3.1.1:C" - <<- [UNSUB 1 ]   <<<why does this happen?>>>
cid:5 - "v3.1.1:C" - Client connection closed: Client Closed
cid:4 - "v1.16.0:go:NATS CLI Version 0.0.33" - Client connection closed: Read Error
```
I think, the highlighted line in the log is (probably) harmless, but still - redundant
It does not happen if I remove `natsSubscription_Destroy` but then I'm leaking memory ofc.
In other words, it looks like the subscription object is not aware that the message limit has been reached and UNSUB is not needed?

thanks in advance!

**Comment by Kazmirchuk** [comment]
same behaviour in the latest version 3.3.0
